src/libutil/util.hh

```diff
@@ -510,6 +510,17 @@ std::optional<N> string2Float(const std::string_view s)
 }
 
 
+/* Convert a little-endian integer to host order. */
+template<typename T>
+T readLittleEndian(unsigned char * p)
+{
+    T x = 0;
+    for (size_t i = 0; i < sizeof(x); ++i)
+        x |= ((T) *p++) << (i * 8);
```

Comment:
```suggestion
    for (size_t i = 0; i < sizeof(x); ++i) {
        x |= ((T) *p) << (i * 8);
        p++;
    }
```

- The braceless `for` loop seems like an easy foot gun (way more than a braceless `if`)
- I'd rather separate the `++` from the rest to make the semantics more obvious